src/main/java/leo/task/Task.java

```diff
@@ -0,0 +1,196 @@
+package leo.task;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.time.format.DateTimeFormatter;
+import java.time.LocalDateTime;
+import java.util.Arrays;
+import java.util.List;
+
+class Task implements Serializable {
+    private static final long serialVersionUID = 1L;
+    
+    enum Type {
+        TODO, DEADLINE, EVENT
+    };
+
+    private String taskDesc;
+    private boolean isDone = false;
+    private Type type;
+    final static List<String> descCommands = Arrays
+            .asList(new String[] { "todo", "deadline", "event", "mark", "unmark", "delete" });
+
+    final static List<String> commands = Arrays
+            .asList(new String[] { "list", "bye", "todo", "deadline", "event", "mark", "unmark", "delete" });
+
+    private Task(String taskDesc, String cmd) {
+        this.taskDesc = taskDesc;
+        this.type = cmd.equals("event") ? Type.EVENT : cmd.equals("deadline") ? Type.DEADLINE : Type.TODO;
+    }
+
+    private char getType() {
+        switch (type) {
+            case DEADLINE:
+                return 'D';
+            case EVENT:
+                return 'E';
+            default:
+                return 'T';
+        }
+    }
+
+    public static Task createTask(String request) throws LeoTaskException {
+        String[] taskArray = request.split(" ", 2);
+
+        try {
+            if (taskArray.length < 2) {
+                if (descCommands.contains(taskArray[0])) {
+                    throw new EmptyFieldException();
+                }
+                throw new InvalidCommandException();
+            }
+            String cmd = taskArray[0], desc = taskArray[1];
+
+            switch (cmd) {
+                case "deadline":
+                    return Deadline.createDeadline(desc, cmd);
+                case "event":
+                    return Event.createEvent(desc, cmd);
+                case "todo":
+                    return new Todo(desc, cmd);
+                default:
+                    throw new InvalidCommandException();
+            }
+        } catch (LeoTaskException e) {
+            e.printStackTrace();
+            System.out.println();
+            return null;
+        }
+    }
+
+    public void setDone() {
+        isDone = true;
+        System.out.println(
+                "Well done on completing the task! Let me mark that as done! Campeon del mundo!");
+        System.out.println(this);
+
+    }
+
+    public void resetDone() {
```

Comment:
The method name resetDone() might be confusing, as it could be interpreted as reset to done. Perhaps setAsUndone() could be used?

Reply:
Updated method name to `setNotDone()`.